moveit_commander/test/python_moveit_commander.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python
+
+import unittest
+import numpy as np
+import rospy
+import rostest
+import os
+
+from moveit_commander import RobotCommander
+
+
+class PythonMoveitCommanderTest(unittest.TestCase):
+    PLANNING_GROUP = "manipulator"
+
+    @classmethod
+    def setUpClass(self):
+        self.commander = RobotCommander("robot_description")
+        self.group = self.commander.get_group(self.PLANNING_GROUP)
+
+    @classmethod
+    def tearDown(self):
+        pass
+
+    def check_target_setting(self, expect, *args):
+        if len(args) == 0:
+            args = [expect]
+        self.group.set_joint_value_target(*args)
+        res = self.group.get_joint_value_target()
+        self.assertTrue(np.all(np.asarray(res) == np.asarray(expect)),
+                        "Setting failed for %s, values: %s" % (type(args[0]), res))
+
+    def test_target_setting(self):
```

Comment:
I think I understand this test but it took me a few minutes to parse what you are doing here. You are testing all the different ways to assign joint values correct? Changing `test_target_setting` to something like `test__group__set_joint_value_target__methods` will go a long way helping future devs debug issues if these tests ever break